src/psyclone/psyGen.py

```diff
@@ -2571,17 +2462,12 @@ def _reduction_ref(self, name):
         to store values into a (padded) array separately for each
         thread.'''
         if self.reprod_reduction:
-            idx_name = self._name_space_manager.create_name(
-                root_name="th_idx",
-                context="PSyVars",
-                label="thread_index")
-            local_name = self._name_space_manager.create_name(
-                root_name="l_"+name,
-                context="PSyVars",
-                label=name)
+            idx_name = \
+                self.root.symbol_table.lookup_tag("omp_thread_index").name
+            local_name = \
+                self.root.symbol_table.name_from_tag(name, "l_" + name)
```

Comment:
Again, can we remove `name_from_tag` here?

Reply:
It won't be easy to replace this instance, I added a TODO for #720 instead